src/data/polynomial/degree/card_pow_degree.lean

```diff
@@ -26,12 +26,15 @@ to `q ^ degree p` (where `q ^ degree 0 = 0`) is an absolute value.
 
 namespace polynomial
 
-variables {Fq : Type*} [field Fq] [fintype Fq]
+variables {Fq : Type*} [fintype Fq]
 
 open absolute_value
 
 open_locale classical polynomial
 
+section
+variables [ring Fq] [no_zero_divisors Fq]
```

Comment:
I'm not sure how valuable this is mathematically; this is finite fields + the zero ring, and on the zero ring we now have the function `\lam 0, 0`. I guess we don't yet have Wedderburn's, but still